pytools/graph.py

```diff
@@ -43,29 +49,46 @@
 Type Variables Used
 -------------------
 
-.. class:: T
+.. class:: NodeT
+
+    Type of a graph node, can be any type.
 
-    Any type.
+.. class:: GraphT
+
+    A :class:`collections.abc.Mapping` representing a directed
+    graph. The mapping contains one key representing each node in the
+    graph, and this key maps to a :class:`collections.abc.Collection` of its
+    successor nodes. Note that most functions expect that every graph node
+    is included as a key in the graph.
 """
 
 from typing import (Collection, TypeVar, Mapping, List, Optional, Any,
-                    Callable, Set, MutableSet, Dict, Iterator, Tuple, FrozenSet)
+                    Callable, Set, MutableSet, Dict, Iterator, Tuple, FrozenSet,
+                    Type)
+
+try:
+    from typing import TypeAlias
+except ImportError:
+    from typing_extensions import TypeAlias
+
+
+NodeT = TypeVar("NodeT")
 
 
-T = TypeVar("T")
+GraphT: TypeAlias = Mapping[NodeT, Collection[NodeT]]
 
 
 # {{{ reverse_graph
 
-def reverse_graph(graph: Mapping[T, Collection[T]]) -> Dict[T, FrozenSet[T]]:
+def reverse_graph(graph: GraphT) -> Dict[NodeT, FrozenSet[NodeT]]:
     """
-    Reverses a graph.
+    Reverses a graph *graph*.
+
+    :arg graph: An instance of :class:`GraphT`.
```

Comment:
```suggestion
```
? The arg doc doesn't bring much now that the type annotation links to the `GraphT` docs.

Reply:
I agree. My thinking was that each argument in the function definition should also be listed here, since otherwise it might look like the two argument lists don't match. I don't have a strong preference either way though.

Reply:
Ok, I removed them in a564ef2